tasks/utils.py

```diff
@@ -133,6 +173,11 @@ def get_build_flags(
         rtloader_headers, rtloader_common_headers
     )
 
+    # adding nikos libs to the env
+    if nikos_embedded_path is not None:
+        env['PKG_CONFIG_PATH'] = env.get('PKG_CONFIG_PATH', '') + ':' + nikos_embedded_path + '/lib/pkgconfig'
+        env["CGO_LDFLAGS"] = env.get('CGO_LDFLAGS', '') + get_nikos_linker_flags(nikos_embedded_path + '/lib')
```

Comment:
Is there a way we could include these flags via `#cgo` comments like this? https://github.com/DataDog/datadog-agent/blob/a6dd21d80946c5f8bc4aa8ef25c5c4671ecbbc77/pkg/ebpf/compiler/compiler.go#L6

Reply:
I tried this just to make sure it would work before answering, and for some reason setting the identical linker flags in a #cgo comment instead of here lead to some linker errors (some undefined reference errors & one `error adding symbols: DSO missing from command line` error). These are fixable, so I do still think it would be possible - however, I don't really think it's worth doing.
To make sure that we're building against a statically compiled, specific version of each dependent library, all of the linker flags are in the form `/opt/nikos/embedded/lib/ + library name + .a`. As a result, the #cgo comment which would replace these flags is super verbose & it's hard to see what's actually being included. Also, it has to be placed in a file within the datadog-agent, since adding it to the source files in nikos would make it very difficult to anyone other than us to build/use nikos. This means we'd be throwing a #cgo comment and including "C" into a file which contains no other cgo & isn't associated with any C source files, which works but it rather obfuscates the build process IMO. Finally, it means that the path to the libraries will always be hardcoded, whereas now the linker flags are generated at runtime based on whatever value is passed to the task with `nikos_embedded_path`.

Reply:
Have you tried specifying the libraries in the form of `-ldnf` (or even `-ldnf.a` if the `.a` extension is required) and then having a single `-L/opt/nikos/embedded/lib`? That should be sufficient. 

> since adding it to the source files in nikos would make it very difficult for anyone other than us to build/use nikos.

Having these in nikos itself, sounds like the correct place for them since they are required to build nikos. If other folks are trying to build from a different directory, they can always use the environment variables to specify alternative directories.

> path to the libraries will always be hardcoded
We hardcode the path to other libraries, so I don't see this as a big deal. As mentioned above, they can use env vars to override.

Reply:
We chatted offline and are going to tackle this in a future PR.